### PR TITLE
Fixed #25764 -- Added support for serialization of enum.Enum in migrations.

### PR DESCRIPTION
**Pull request by andreif**
Ticket https://code.djangoproject.com/ticket/25764

### CODE REVIEW

File: docs/internals/contributing/writing-code/unit-tests.txt [hunk 1]
```diff
@@ -138,6 +138,7 @@ dependencies:
 
 *  bcrypt_
 *  docutils_
+*  enum34_ 2.7-3.3
```
Comment (timgraham) [inline docs/internals/contributing/writing-code/unit-tests.txt]: 2.7+ below refers to the version of jinja2, we can change this to "(Python 2 only)"

File: docs/releases/1.10.txt [hunk 2]
```diff
@@ -165,7 +165,10 @@ Management Commands
 Migrations
 ^^^^^^^^^^
 
-* ...
+* Serialize ``enum.Enum`` instances (native 3.4+ and from enum34_ 2.7-3.3) so
```
Comment (timgraham) [inline docs/releases/1.10.txt]: I think we can remove the parenthetical reference (same in topics guide). Django 1.10 only supports Python 3.4+ so this is only relevant for Python 2.7 and I think it's reasonable that readers will assume the backported version works the same.

File: tests/migrations/test_writer.py [hunk 3]
```diff
@@ -229,6 +234,40 @@ def test_serialize_builtin_types(self):
...
+    def test_serialize_enums(self):
+        class Enum(enum.Enum):
+            A = 'a-value'
+            B = 'value-b'
+
+        class IntEnum(enum.IntEnum):
+            A = 1
+            B = 2
+
+        self.assertSerializedResultEqual(
```
Comment (timgraham) [inline tests/migrations/test_writer.py]: Like this:
```
self.assertSerializedResultEqual(
    Enum.A,
    ("migrations.test_writer.Enum('a-value')", {'import migrations.test_writer'})
)
```
(makes it easier to track the closing parenthesis in my opinion)

**Comment by andreif** [comment]
@timgraham fixed

**Comment by andreif** [comment]
(removed remaining reference)

**Comment by timgraham** [comment]
Some minor edits: http://dpaste.com/3AXZQX0

Upon installing enum34 on Python 2, I noticed the test doesn't pass there.

**Comment by timgraham** [comment]
Is it possible to use a similar fix?
```
if six.PY2:
    # Strip the `u` prefix since we're importing unicode_literals
    value_repr = value_repr[1:]
```

**Comment by andreif** [comment]
Yeah, trying to apply it. I think I should serialize the enum member value as well, working on it

**Comment by andreif** [comment]
@timgraham it seems that jenkins is hanging, I'll try again

**Comment by andreif** [comment]
@timgraham Jenkins seems randomly hang or fail for Windows

**Comment by timgraham** [comment]
It's under a lot of load now. Try to avoid pushing your branch a lot (e.g. pushing a new commit, then immediately pushing a squashed one) since this will trigger two builds.

**Comment by andreif** [comment]
@timgraham Is there a way to trigger tests without pushing?

**Comment by timgraham** [comment]
See https://code.djangoproject.com/wiki/Jenkins but there's no need to retrigger just for random failures.

**Comment by timgraham** [comment]
merged in 998894e1b9f5a8715f4cac5f6201b5e9ac80510c, thanks!

**Pull request closed without merging**